SwiftOverpass/Queries/OverpassQuery.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  OverpassQuery.swift
+//  SwiftOverpass
+//
+//  Created by Sho Kamei on 2017/12/03.
+//  Copyright © 2017年 Sho Kamei. All rights reserved.
+//
+
+import Foundation
+import AEXML
+
+// Represents types of queries.
+public enum OverpassQueryType {
+    /// Specifies the query is node
+    case node
+    /// Specifies the query is way
+    case way
+    /// Specifies the query is relation
+    case relation
+    
+    /// The string to creat a XML line.
+    public var stringValue: String {
+        return OverpassQueryType.stringMapping[self]!
+    }
+    
+    fileprivate static let stringMapping = [
+        node: "node",
+        way: "way",
+        relation: "relation"
+    ]
+}
+
+/**
+*/
+public protocol OverpassQuery: class {
+    var type: OverpassQueryType { get }
+    var parent: OverpassQuery? { get }
+    var tags: [String: OverpassTag] { get set }
+    var boundingBox: BoudingBox? { get set }
+}
+
+extension OverpassQuery {
+    /**
+     Sets a bbox to retrieve from specific location
+     
+     - parameter bbox: A bouding box to set
+     */
+    public func setBoudingBox(_ bbox: BoudingBox) {
+        boundingBox = bbox
+    }
+    
+    /**
+     Sets a bbox to retrieve from specific location
+     
+     - parameter s: Lowest latitude
+     - parameter n: Lowest longitude
+     - parameter w: Highest latitude
+     - parameter e: Highest longitude
+     */
+    public func setBoudingBox(s: Double, n: Double, w: Double, e: Double) {
```

Comment:
To me as someone not (yet) using this part of the framework, it's a bit confusing to have two methods to set the bounding box. Is this for convenience reasons?

When setting the bounding box of a query, I'd write something along
```
let query = ...

query.boundingBox = BoundingBox(...)
```

Do you feel it's necessary to have these two methods? If we wanted to cover `OverpassQuery` with unit tests, we'd need to have one test for each of these, increasing the effort.

I suggest we remove these two methods and just use the
```
var boundingBox: BoudingBox? { get set }
```

Reply:
Yes, you are right. It will make people confused. There are no reasons for those two methods. So I will edit the code along with your suggestion!